### DISCUSSION TITLE
Tasmota ESP32-Cam + Temp DS18B20 + External Light

### DISCUSSION THREAD

**Discussion by mischa-t** [arendst/Tasmota#18541] [Q&A]
Hello, i have a new Project with an ESP32-Cam Board.
I want to Flash Tosmota to the Board and additional hook up a Temp Sensor (DS1820) and an External light.
I would like to let the Camera make Fotos by movement on the SD Card, additional over the Webserver GUI i would like to se the Temp from the external sensor and also Switch an external light on if required.
With the 2 last item ( External Temp indication & Seperate light On/OFF) i have aproblem and dont know how.
Does anbody have a tip and can help.

**Comment by Jason2866** [comment]
The Webcam env build does not have support for sensors. Webcam task is already a heave task for the esp32

**Comment by mischa-t** [comment]
ok, thanks for the info.

**Reply by kodacy** [reply]
A switch and a temperature sensor are not heavy tasks, but as far as i know you will need to compile tasmota youself: you can find a guide here on how to add the Temp/Hum sensor to the firmware: https://cgomesu.com/blog/Esp32cam-tasmota-webcam-server/#fixing-the-timezone.
